modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/util/JournalConverterImpl.java

```diff
@@ -181,13 +190,24 @@ private void _addDDMFields(
 		for (Element dynamicElementElement : dynamicElementElements) {
 			if (!ddmFormField.isTransient()) {
 				Field ddmField = _getField(
-					dynamicElementElement, ddmStructure, availableLanguageIds,
-					defaultLanguageId);
-
-				String fieldName = ddmField.getName();
+					availableLanguageIds, ddmStructure, defaultLanguageId,
+					dynamicElementElement, fieldName);
 
 				Field existingDDMField = ddmFields.get(fieldName);
 
+				if (existingDDMField == null) {
+					String legacyFieldName =
+						DDMFormFieldUtil.getLegacyFieldName(fieldName);
```

Comment:
If we change the method name, then I would rename this variable to `legacyDDMFormFieldName`